waspc/src/Wasp/Analyzer/Evaluator/Evaluation/TypedExpr/Combinators.hs

```diff
@@ -85,6 +88,54 @@ list elemEvaluation = evaluation $ \(typeDefs, bindings) -> \case
       mapM (runEvaluation elemEvaluation typeDefs bindings) values
   expr -> Left $ EvaluationError.ExpectedListType $ TypedAST.exprType expr
 
+-- | An evaluation that expects a "Tuple" with 2 elements (pair) and runs the
```

Comment:
These 3 functions are repetitive / boilerplate, and are a candidate for generating via Template Haskell (TH).
I implemented such function in TH that would generate these functions, however at the end I concluded it is not worth the complexity of maintaining it, since TH is relatively complex, and we only have tuples from 2 to 4 in length -> I doubt we will ever have longer tuples. If we instead had tuples from 2 to 8 in length, or even more, I would certainly advise we do use TH, but I believe that will never be the case.